stdlib/SparseArrays/src/sparsevector.jl

```diff
@@ -1935,6 +1935,16 @@ julia> dropzeros(A)
 """
 dropzeros(x::SparseVector; trim::Bool = true) = dropzeros!(copy(x), trim = trim)
 
+function copy!(dst::SparseVector, src::SparseVector)
+    dst.n == src.n || throw(ArgumentError("Vector should have the same length for copy!"))
```

Comment:
For `AbstractVector` in general, `dst` and `src` can be of different lengths, I don't see here the necessity to have this restriction. I think the two calls to `copy!` below will handle different lengths just fine.

Reply:
but `SparseVector` is immutable, so tere is no way of changing `n`. Should `SparseVector` be mutable?

Reply:
Yes sorry, I overlooked that! Then just update the error message to tell `"Sparse vectors should..."` to emphasize that it concerns only `SparceVector`.